lib/config/Configuration.js

```diff
@@ -0,0 +1,319 @@
+import path from "node:path";
+import os from "node:os";
+
+/**
+ *
+ *
+ * @public
+ * @class
+ * @alias @ui5/project/config/Configuration
+ */
+export class Configuration {
```

Comment:
IMHO we could put the configuration feature in a separate PR, merge that one first and the rebase the snapshot PR.

Reply:
With that we make this PR here not even larger.

Reply:
Moved to a separate change: https://github.com/SAP/ui5-project/pull/575